tensorflow/core/kernels/cwise_op_gpu_select.cu.cc

```diff
@@ -33,6 +33,20 @@ struct SelectFunctor<GPUDevice, T> {
 };
 
 template <typename T>
+struct SelectScalarFunctor<GPUDevice, T> {
+  void operator()(const GPUDevice& d, typename TTypes<T>::Flat out,
+                  typename TTypes<bool>::ConstScalar cond,
+                  typename TTypes<T>::ConstFlat then_flat,
+                  typename TTypes<T>::ConstFlat else_flat) {
+    if (cond()){
```

Comment:
even that kernel has a comment from @ebrevdo that leads to erroneous memory access. Seems something pesky is going on underneath in Eigen.
The second solution to just pass a bool to the kernel. But I wonder if that would require CPU sync with the op to transfer that single boolean. @benoitsteiner Do you have any insight in why a call to a ScalarBool on gpu leads to memory access errors ?